src/hardware/setup.rs

```diff
@@ -699,31 +689,25 @@ pub fn setup(
         unsafe { ethernet::enable_interrupt() };
 
         // Configure IP address according to DHCP socket availability
-        let ip_addrs: smoltcp::wire::IpCidr;
-        if NUM_DHCP_SOCKETS == 1 {
-            ip_addrs = smoltcp::wire::IpCidr::new(
-                smoltcp::wire::IpAddress::Ipv4(
-                    smoltcp::wire::Ipv4Address::UNSPECIFIED,
-                ),
-                0,
-            );
-        } else {
-            ip_addrs = smoltcp::wire::IpCidr::new(
-                smoltcp::wire::IpAddress::Ipv4(
-                    smoltcp::wire::Ipv4Address::new(
-                        IPADDR[0], IPADDR[1], IPADDR[2], IPADDR[3],
-                    ),
-                ),
-                24,
-            );
-        }
+        let ip_addrs: smoltcp::wire::IpAddress = option_env!("STATIC_IP")
+            .unwrap_or("UNSPECIFIED")
+            .parse()
+            .unwrap();
 
         // Note(unwrap): The hardware configuration function is only allowed to be called once.
         // Unwrapping is intended to panic if called again to prevent re-use of global memory.
         let store =
             cortex_m::singleton!(: NetStorage = NetStorage::default()).unwrap();
 
-        store.ip_addrs[0] = ip_addrs;
+        store.ip_addrs[0] = match ip_addrs.is_unspecified() {
```

Comment:
If ip_addrs is already unspecified, why do we need to construct it with a different netmask below? Can't both use the same 24 bit netmask without issue?